sphinx_gallery/docs_resolv.py

```diff
@@ -273,7 +276,12 @@ def resolve(self, cobj, this_url, return_type=False):
         full_name = cobj['module_short'] + '.' + cobj['name']
         if full_name not in self._link_cache:
             # we don't have it cached
-            self._link_cache[full_name] = self._get_link_type(cobj)
+            pattern = 'osaft.solutions.\w+\d{4}\w*.ARF$'
```

Comment:
Here would need to stand the `prefere_full_module` option or similar. I just did like that for now to see if it would general work and to have a discussion basis.